lib/ansible/module_utils/hwc_utils.py

```diff
@@ -0,0 +1,324 @@
+            expect_vals[k] = v
+
+    return DictComparison(expect_vals) != DictComparison(actual_vals)
+
+
+class HwcSession(object):
+    """Handles all authentation and HTTP sessions for HWC API calls."""
+
+    def __init__(self, module, product):
+        self.module = module
+        self.product = product
+        self._validate()
+        self._session = self._credentials()
+        self._adapter = Adapter(self._session)
+        self._endpoints = {}
+        self._project_id = ""
+
+    def get(self, url, body=None):
+        try:
+            return self._adapter.get(
+                url, json=body,
+                headers=self._headers(), raise_exc=False)
+        except getattr(requests.exceptions, 'RequestException') as inst:
+            self.module.fail_json(msg=inst.message)
+
+    def post(self, url, body=None):
+        try:
+            return self._adapter.post(
+                url, json=body,
+                headers=self._headers(), raise_exc=False)
+        except getattr(requests.exceptions, 'RequestException') as inst:
+            self.module.fail_json(msg=inst.message)
+
+    def delete(self, url, body=None):
+        try:
+            return self._adapter.delete(
+                url, json=body,
+                headers=self._headers(), raise_exc=False)
+        except getattr(requests.exceptions, 'RequestException') as inst:
+            self.module.fail_json(msg=inst.message)
+
+    def put(self, url, body=None):
+        try:
+            return self._adapter.put(
+                url, json=body,
+                headers=self._headers(), raise_exc=False)
+        except getattr(requests.exceptions, 'RequestException') as inst:
+            self.module.fail_json(msg=inst.message)
+
+    def get_service_endpoint(self, service_type):
+        if self._endpoints.get(service_type):
+            return self._endpoints.get(service_type)
+
+        e = None
+        try:
+            e = self._session.get_endpoint_data(
+                service_type=service_type,
+                region_name=self.module.params['region']
+            )
+        except getattr(requests.exceptions, 'RequestException') as inst:
+            self.module.fail_json(msg=inst.message)
+
+        if not e or e.url == "":
+            self.module.fail_json(
+                msg="Can not find the enpoint for %s" % service_type)
+
+        url = e.url
+        if url[-1] != "/":
+            url += "/"
+
+        self._endpoints[service_type] = url
+        return url
+
+    def get_project_id(self):
+        if self._project_id:
+            return self._project_id
+        try:
+            pid = self._session.get_project_id()
+            self._project_id = pid
+            return pid
+        except getattr(requests.exceptions, 'RequestException') as inst:
+            self.module.fail_json(msg=inst.message)
+
+    def _validate(self):
+        if not HAS_REQUESTS:
+            self.module.fail_json(msg=missing_required_lib('requests'),
```

Comment:
use basic.py's `missing_required_lib` function as it gives details on host and python to avoid user confusion